compiler/executable/match_/planner/vertex/variable.rs

```diff
@@ -52,39 +49,49 @@ impl VariableVertex {
         }
     }
 
+    pub(crate) fn set_binding(&mut self, binding_pattern: PatternVertexId) {
+        match self {
+            Self::Input(_) => unreachable!("attempting to assign to input variable"),
```

Comment:
Is binding defined as in the spec? Or Is it only for some specific variables?
If so, Can't you bind an input variable with a new constraint?

```
match $x has $y;
match $x isa person;
```
?

Reply:
This sets whether the variable is assigned to by another pattern, as we can't produce it in another way.
I guess `set_assignment_source` would be a better name?

Reply:
I'll ask you this in person. We probably catch this earlier but no harm in having a check here as well. It might even be the right place to do it.